backend/src/schema/resolvers/inviteCodes.js

```diff
@@ -0,0 +1,109 @@
+import generateInvieCode from './helpers/generateInviteCode'
+import Resolver from './helpers/Resolver'
+
+const uniqueInviteCode = async (session, code) => {
+  return session.readTransaction(async (txc) => {
+    const result = await txc.run(`MATCH (ic:InviteCode { id: $code }) RETURN count(ic) AS count`, {
+      code,
+    })
+    return parseInt(String(result.records[0].get('count'))) === 0
+  })
+}
+
+export default {
+  Query: {
+    MyInviteCodes: async (_parent, args, context, _resolveInfo) => {
+      const {
+        user: { id: userId },
+      } = context
+      const session = context.driver.session()
+      const readTxResultPromise = session.readTransaction(async (txc) => {
+        const result = await txc.run(
+          `MATCH (user:User {id: $userId})-[:GENERATED]->(ic:InviteCode)
+           RETURN properties(ic) AS inviteCodes`,
+          {
+            userId,
+          },
+        )
+        return result.records.map((record) => record.get('inviteCodes'))
+      })
+      try {
+        const txResult = await readTxResultPromise
+        return txResult
+      } finally {
+        session.close()
+      }
+    },
+    isValidInviteCode: async (_parent, args, context, _resolveInfo) => {
+      const { code } = args
+      if (!code) return false
+      const session = context.driver.session()
+      const readTxResultPromise = session.readTransaction(async (txc) => {
+        const result = await txc.run(
+          `MATCH (ic:InviteCode { code: $code })
```

Comment:
The user input should be caps-agnostic - so Something like this should do the trick (Pseudocode, not tested)

```suggestion
          `MATCH (ic:InviteCode { code: UPPER($code) })
```

Reply:
Great. Thank you!